server/client.go

```diff
@@ -1446,26 +1477,37 @@ func (c *client) flushOutbound() bool {
 		return true // true because no need to queue a signal.
 	}
 
-	// Place primary on nb, assign primary to secondary, nil out nb and secondary.
-	nb, attempted := c.collapsePtoNB()
-	c.out.p, c.out.nb, c.out.s = c.out.s, nil, nil
-	if nb == nil {
-		return true
-	}
-
-	// For selecting primary replacement.
-	cnb := nb
-	var lfs int
-	if len(cnb) > 0 {
-		lfs = len(cnb[0])
-	}
+	// In the case of a normal socket connection, "collapsed" is just a ref
+	// to "nb". In the case of WebSockets, additional framing is added to
+	// anything that is waiting in "nb". Also keep a note of how many bytes
+	// were queued before we release the mutex.
+	collapsed, attempted := c.collapsePtoNB()
+
+	// Frustratingly, (net.Buffers).WriteTo() modifies the receiver so we
+	// can't work on "nb" directly — while the mutex is unlocked during IO,
+	// something else might call queueOutbound and modify it. So instead we
+	// need a working copy — we'll operate on "wnb" instead. Note that in
```

Comment:
Can this just be on the stack vs part of client?

Reply:
The reason for having it on the client is that we can reuse the underlying array by just trimming it down with `wnb[:0]` and store it for next time around which saves on some more allocations.

Reply:
So we buffer in two spots, pool and wnb?

Reply:
Can these be cleaned up if client goes back to idle?

Reply:
`wnb` is reused each time `flushOutbound` calls. Actually it's a good point that right now we don't check if the capacity of `wnb` has grown uncontrollably so I'll add something to check for that.